app/assets/javascripts/code/services/_sensors.js

```diff
@@ -163,3 +148,76 @@ const max = (valueOf, xs) => {
   const reducer = (acc, x) => valueOf(x) > valueOf(acc) ? x : acc;
   return xs.reduce(reducer, xs[0]);
 };
+
+export const findAvailableSensorsForParameter = (sort, sensors, parameter) =>
+  parameter ?
+    sort(Object.values(sensors).filter(sensor => sensor.measurement_type === parameter.id)) :
+    sort(Object.values(sensors));
+
+export const sort = sensors => {
+  const ORDERS = {
+    "Particulate Matter-AirBeam2-PM2.5 (µg/m³)": 4,
+    "Particulate Matter-AirBeam2-PM1 (µg/m³)": 3,
+    "Particulate Matter-AirBeam2-PM10 (µg/m³)": 2,
+    "Particulate Matter-AirBeam-PM (µg/m³)": 1,
+    "Humidity-AirBeam2-RH (%)": 2,
+    "Humidity-AirBeam-RH (%)": 1,
+    "Temperature-AirBeam2-F (F)": 2,
+    "Temperature-AirBeam-F (F)": 1,
+    "Sound Level-Phone Microphone (dB)": 1
+  };
+
+  const compare = (sensor1, sensor2) => {
+    if (ORDERS[sensor1.id] && ORDERS[sensor2.id]) {
+      return ORDERS[sensor2.id] - ORDERS[sensor1.id];
+    } else if (ORDERS[sensor1.id]) {
+      return -1;
+    } else if (ORDERS[sensor2.id]) {
+      return +1;
+    } else {
+      return sensor2.session_count - sensor1.session_count;
+    }
+  }
+
+  return sensors.sort(compare);
+}
+
+const buildSensorId = sensor =>
+  sensor.measurement_type + "-" + sensor.sensor_name + " (" + sensor.unit_symbol + ")";
+
+export const defaultSensorIdForParameter = (parameter, sensors) => {
+  const DEFAULT_IDS = {
+    "Particulate Matter": buildSensorId({
+      measurement_type: "Particulate Matter",
+      sensor_name:      "AirBeam2-PM2.5",
+      unit_symbol:      "µg/m³"
+    }),
+    "Humidity": buildSensorId({
+      measurement_type: "Humidity",
+      sensor_name:      "AirBeam2-RH",
+      unit_symbol:      "%"
+    }),
+    "Temperature": buildSensorId({
+      measurement_type: "Temperature",
+      sensor_name:      "AirBeam2-F",
+      unit_symbol:      "F"
+    }),
+    "Sound Level": buildSensorId({
+      measurement_type: "Sound Level",
+      sensor_name:      "Phone Microphone",
+      unit_symbol:      "dB"
+    })
+  };
+
+  return DEFAULT_IDS[parameter.id] || max(sensor => sensor.session_count, sensors).id;
+};
+
+export const buildAvailableParameters = sensors => {
+  const uniq = (v, i, a) => a.indexOf(v) === i;
```

Comment:
Does v, i, a stand for something? If yes then could you use the full word? If they are meant to be generic then why not a, b, c? 
Also, is this the same as in line 219?

Reply:
They stand for value, index, array. I totally agree with you but this time I'll leave that for later. Again, I've refactored a lot and don't want to spend more time on cosmetics for this round